File: LeetCode/Easy/0278-first-bad-version/0278-first-bad-version.go
/** 
 * Forward declaration of isBadVersion API.
 * @param   version   your guess about first bad version
 * @return 	 	      true if current version is bad 
 *			          false if current version is good
 * func isBadVersion(version int) bool;
 */

func firstBadVersion(n int) int {
    good_max := 0
    bad_min := n
    mid := (good_max + bad_min) / 2
    for good_max + 1 != bad_min {
        if isBadVersion(mid) {
            bad_min = mid
        } else {good_max = mid}
        mid = (good_max + bad_min) / 2
    }
    return bad_min
}

File: LeetCode/Easy/0278-first-bad-version/0278-first-bad-version.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        long good_max = 0;
        long bad_min = n;
        long mid = (good_max + bad_min) / 2;
        while (good_max + 1 != bad_min){
            if (isBadVersion((int) mid)) bad_min = mid;
            else good_max = mid;
            mid = (good_max + bad_min) / 2;
        }
        return (int) bad_min;
    }
}